Mathlib/FieldTheory/NormalClosure.lean

```diff
@@ -73,12 +82,74 @@ instance is_finiteDimensional [FiniteDimensional F K] :
   haveI : ∀ f : K →ₐ[F] L, FiniteDimensional F f.fieldRange := fun f =>
     f.toLinearMap.finiteDimensional_range
   apply IntermediateField.finiteDimensional_iSup_of_finite
-#align normal_closure.is_finite_dimensional normalClosure.is_finiteDimensional
 
-instance isScalarTower : IsScalarTower F (normalClosure F K L) L :=
-  -- Porting note: the last argument here `(⨆ (f : K →ₐ[F] L), f.fieldRange).toSubalgebra`
-  -- was just written as `_` in mathlib3.
-  IsScalarTower.subalgebra' F L L (⨆ (f : K →ₐ[F] L), f.fieldRange).toSubalgebra
-#align normal_closure.is_scalar_tower normalClosure.isScalarTower
+noncomputable instance algebra : Algebra K (normalClosure F K L) :=
+  ((inclusion (toAlgHom F K L).fieldRange_le_normalClosure).comp
+    (toAlgHom F K L).rangeRestrict).toAlgebra
```

Comment:
This almost certainly creates a diamond that didn't exist before: `RingHom.toAlgebra` is almost always a bad idea.
You can fix it with
```suggestion
noncomputable instance algebra : Algebra K (normalClosure F K L) :=
  IntermediateField.algebra <|
    { (⨆ f : K →ₐ[F] L, f.fieldRange).toSubfield with
      -- Porting note: could not inherit neg_mem
      neg_mem' := fun _ hx => Subfield.neg_mem (⨆ f : K →ₐ[F] L, f.fieldRange).toSubfield hx
      algebraMap_mem' := fun r =>
        le_iSup (fun f : K →ₐ[F] L => f.fieldRange) (IsScalarTower.toAlgHom F K L) ⟨r, rfl⟩ }
```
but this seems to defeat the point of the refactor.

Reply:
Well, the point of the refactor is to make `normalClosure` easier to use, so it's not a big deal that the algebra instance retains the messiness of the original definition.

Reply:
I pushed your definition (slightly golfed)